### PR TITLE
Allow to update to unstable releases

### PR DESCRIPTION
**Pull request by skjnldsv**
```
$ docker-compose exec --user=docker php php occ app:remove forms               
forms disabled
forms 1.1.1 removed

$ docker-compose exec --user=docker php php occ app:install forms              
forms 1.1.1 installed
forms enabled

$ docker-compose exec --user=docker php php occ app:update forms --allownightly
forms new version available: 2.0.0-beta1
forms updated
```

### CODE REVIEW

File: core/Command/App/Disable.php [hunk 1]
```diff
@@ -78,7 +78,8 @@ private function disableApp(string $appId, OutputInterface $output): void {
 
 		try {
 			$this->appManager->disableApp($appId);
-			$output->writeln($appId . ' disabled');
+			$appVersion = \OC_App::getAppVersion($appId);
+			$output->writeln($appId . ' ' . $appVersion . ' disabled');
```
Comment (MorrisJobke) [inline core/Command/App/Disable.php:82]: 😍

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍 

I'm a bit undecided whether the name allow nightly vs allow unstable is a good decision. We should make it easier to our future self to properly understand this and having "allow nightly" also whitelist beta releases is a bit weird. Having the CLI parameter being the same as the internal variable makes it a bit more obvious IMO.

File: core/Command/App/Update.php [hunk 2]
```diff
@@ -76,7 +76,12 @@ protected function configure() {
 				InputOption::VALUE_NONE,
 				'show update(s) without updating'
 			)
-
+			->addOption(
+				'allownightly',
```
Comment (ChristophWurst) [inline core/Command/App/Update.php:80]:
```suggestion
				'allow-nightly',
```

:shrug:?

**Review by ChristophWurst — APPROVED** [review]

File: core/Command/App/Update.php (hunk 2 above)
Comment (MorrisJobke) [inline core/Command/App/Update.php:80]:
```suggestion
				'allow-unstable',
```
🤔

File: core/Command/App/Update.php [hunk 3]
```diff
@@ -76,7 +76,12 @@ protected function configure() {
...
+				null,
+				InputOption::VALUE_NONE,
+				'allow updating to unstable releases'
```
Comment (MorrisJobke) [inline core/Command/App/Update.php:83]: Also inlines the name with the description

**Comment by skjnldsv** [comment]
> I'm a bit undecided whether the name allow nightly vs allow unstable is a good decision. We should make it easier to our future self to properly understand this and having "allow nightly" also whitelist beta releases is a bit weird. Having the CLI parameter being the same as the internal variable makes it a bit more obvious IMO.

Same, I'm not super sure!

File: core/Command/App/Update.php (hunk 2 above)
Comment (skjnldsv) [reply core/Command/App/Update.php:80]: Sure! The one above was called `showonly`, so I figured i'd be consistent on the casing :shrug: 
cc @ChristophWurst

**Comment by skjnldsv** [comment]
> also whitelist beta releases is a bit weird

Btw, it's also weird as we don't have a way to send beta to the appstore, so i don't understand why we have three settings instead of just having like prod vs beta-alpha :thinking:

File: core/Command/App/Disable.php (hunk 1 above)
Comment (skjnldsv) [reply core/Command/App/Disable.php:82]: Yes, this was bothering me.. Stupid but really great to have ^^'

**Comment by szaimen** [comment]
I think it would be great if there were two dlags: one to install the latest beta release and another to install the latest nightly release.
Because it could happen that an app offers a beta release as well as a nightly release at the same time and the admin would be able to choose which one he/she would like to install because for some admins maybe a nightly is too risky on a productive instance but beta okay...

File: core/Command/App/Update.php (hunk 2 above)
Comment (ChristophWurst) [reply core/Command/App/Update.php:80]: I checked a few other commands and the `-` casing is used elsewhere as well

Comment (skjnldsv) [reply core/Command/App/Update.php:80]: Let's go for this then!
Thanks :)

**Comment by skjnldsv** [comment]
> I think an update option to update to a specific version of an app is the right direction but it would be great if there were two options: one to install the latest beta version and another to install the latest nightly version.

I would say this is a secondary step :)

**Comment by szaimen** [comment]
> I would say this is a secondary step :)
> 
> The appstore only have a nightly option 🤔

I guess then it is as good as it gets as long as the appstore only has a nightly option...

**Review by szaimen — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
I'm fine with *allow-unstable* as we do not have a stability policy for occ and thus can adjust this in the future if necessary.

**Comment by MorrisJobke** [comment]
CI failures .7

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Please also update the documentation then as well :)

**Comment by MorrisJobke** [comment]
Unit tests fail

**Comment by blizzz** [comment]
what's the state here?

**Comment by skjnldsv** [comment]
Fixed tests & rebased

**Review by MorrisJobke — APPROVED** [review]
Code looks good 👍

**Pull request merged**